extensions/standard-processors/processors/SegmentContent.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SegmentContent.h"
+
+#include "core/FlowFile.h"
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+#include "core/Resource.h"
+#include "range/v3/view/split.hpp"
+#include "utils/ProcessorConfigUtils.h"
+
+namespace org::apache::nifi::minifi::processors {
+
+constexpr size_t BUFFER_TARGET_SIZE = 1024;
+
+void SegmentContent::initialize() {
+  setSupportedProperties(Properties);
+  setSupportedRelationships(Relationships);
+}
+
+void SegmentContent::onSchedule(core::ProcessContext&, core::ProcessSessionFactory&) {}
+
+namespace {
+std::shared_ptr<core::FlowFile> createSegment(core::ProcessSession& session) {
+  auto first_split = session.create();
+  if (!first_split) { throw Exception(PROCESSOR_EXCEPTION, "Couldn't create FlowFile"); }
+  return first_split;
+}
+
+void updateSplitAttributesAndTransfer(core::ProcessSession& session, const std::vector<std::shared_ptr<core::FlowFile>>& splits, const core::FlowFile& original) {
+  const std::string fragment_identifier_ = utils::IdGenerator::getIdGenerator()->generate().to_string();
```

Comment:
Why don't we use the unique ID of the parent flow file? I think that would be helpful to users.

Reply:
As we discussed offline, I think it would make sense to diverge from NiFi in this respect because in MiNiFi we normally don't use provenance which would be the other way to connect the fragments to the original.

But if you prefer to follow the behavior of NiFi more closely, that makes sense, too. I'll approve the PR in either case.

Reply:
I went with your idea, added some tests aswell https://github.com/apache/nifi-minifi-cpp/pull/1879/commits/b78de39d16432f40be3cf18a76d9005b79ffdb92